`TestLoginRuleController` flakiness

## Failure

`TestLoginRuleController` is flaky since the introduction of the race detector in e/.

#### Link(s) to logs

- https://github.com/gravitational/teleport.e/actions/runs/5048568926/jobs/9056926772?pr=1470

#### Relevant snippet

```
go test -v ./lib/operatortest -race
=== RUN   TestLoginRuleController
=== PAUSE TestLoginRuleController
=== CONT  TestLoginRuleController
time="2023-05-22T17:18:3[9](https://github.com/gravitational/teleport.e/actions/runs/5048568926/jobs/9056926772?pr=1470#step:9:10)Z" level=info msg="Generating TLS certificate 1.3.9999.1.7=#13096c6f63616c686f7374,CN=test-tls-server.localhost,O=Auth,POSTALCODE=null,STREET=" dns_names="[*.teleport.cluster.local teleport.cluster.local]" key_usage=5 not_after="2033-05-19 17:18:39.222604203 +0000 UTC" trace.component=ca
time="2023-05-22T17:18:39Z" level=info msg="Generating TLS certificate 1.3.9999.1.7=#13096c6f63616c686f7374,CN=Admin.localhost,O=Admin,POSTALCODE=null,STREET=" dns_names="[*.teleport.cluster.local teleport.cluster.local]" key_usage=5 not_after="2033-05-19 17:18:39.222604203 +0000 UTC" trace.component=ca
=== RUN   TestLoginRuleController/creation
=== PAUSE TestLoginRuleController/creation
=== RUN   TestLoginRuleController/deletion_drift
=== PAUSE TestLoginRuleController/deletion_drift
=== RUN   TestLoginRuleController/update
=== PAUSE TestLoginRuleController/update
=== CONT  TestLoginRuleController/update
=== CONT  TestLoginRuleController/deletion_drift
=== CONT  TestLoginRuleController/creation
==================
WARNING: DATA RACE
Write at 0x00c00085f3e0 by goroutine 91:
  runtime.mapassign()
      /opt/go/src/runtime/map.go:578 +0x0
  k8s.io/apimachinery/pkg/runtime.(*Scheme).AddUnversionedTypes()
      /go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/runtime/scheme.go:[12](https://github.com/gravitational/teleport.e/actions/runs/5048568926/jobs/9056926772?pr=1470#step:9:13)9 +0x58b
  k8s.io/apimachinery/pkg/apis/meta/v1.AddToGroupVersion()
      /go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/apis/meta/v1/register.go:75 +0x60a
  sigs.k8s.io/controller-runtime/pkg/scheme.(*Builder).Register.func1()
      /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.[14](https://github.com/gravitational/teleport.e/actions/runs/5048568926/jobs/9056926772?pr=1470#step:9:15).6/pkg/scheme/scheme.go:72 +0xbb
  k8s.io/apimachinery/pkg/runtime.(*SchemeBuilder).AddToScheme()
      /go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/runtime/scheme_builder.go:29 +0xa1
  sigs.k8s.io/controller-runtime/pkg/scheme.(*Builder).AddToScheme()
      /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/scheme/scheme.go:86 +0x2c
  sigs.k8s.io/controller-runtime/pkg/scheme.(*Builder).AddToScheme-fm()
      <autogenerated>:1 +0x44
  github.com/gravitational/teleport/integrations/operator/controllers/resources/testlib.SetupTestEnv()
      /__w/teleport.e/teleport.e/integrations/operator/controllers/resources/testlib/env.go:262 +0x3b8
  github.com/gravitational/teleport/integrations/operator/controllers/resources/testlib.ResourceDeletionDriftTest[...]()
      /__w/teleport.e/teleport.e/integrations/operator/controllers/resources/testlib/teleport_reconciler_tests.go:89 +0xab
  github.com/gravitational/teleport/integrations/operator/controllers/resources/testlib.LoginRuleDeletionDriftTest()
      /__w/teleport.e/teleport.e/integrations/operator/controllers/resources/testlib/login_rule_controller_tests.go:149 +0xbc
  github.com/gravitational/teleport/e/lib/operatortest.TestLoginRuleController.func1()
      /__w/teleport.e/teleport.e/e/lib/operatortest/loginrule_controller_test.go:43 +0x52
  testing.tRunner()
      /opt/go/src/testing/testing.go:[15](https://github.com/gravitational/teleport.e/actions/runs/5048568926/jobs/9056926772?pr=1470#step:9:16)76 +0x2[16](https://github.com/gravitational/teleport.e/actions/runs/5048568926/jobs/9056926772?pr=1470#step:9:17)
  sigs.k8s.io/controller-runtime/pkg/scheme.(*Builder).Register.func1()
      /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/scheme/scheme.go:72 +0xbb
  k8s.io/apimachinery/pkg/runtime.(*SchemeBuilder).AddToScheme()
      /go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/runtime/scheme_builder.go:29 +0xa1
  sigs.k8s.io/controller-runtime/pkg/scheme.(*Builder).AddToScheme()
      /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/scheme/scheme.go:86 +0x2c
  sigs.k8s.io/controller-runtime/pkg/scheme.(*Builder).AddToScheme-fm()
      <autogenerated>:1 +0x44
  github.com/gravitational/teleport/integrations/operator/controllers/resources/testlib.SetupTestEnv()
      /__w/teleport.e/teleport.e/integrations/operator/controllers/resources/testlib/env.go:271 +0x624
  github.com/gravitational/teleport/integrations/operator/controllers/resources/testlib.ResourceDeletionDriftTest[...]()
      /__w/teleport.e/teleport.e/integrations/operator/controllers/resources/testlib/teleport_reconciler_tests.go:89 +0xab
  github.com/gravitational/teleport/integrations/operator/controllers/resources/testlib.LoginRuleDeletionDriftTest()
      /__w/teleport.e/teleport.e/integrations/operator/controllers/resources/testlib/login_rule_controller_tests.go:149 +0xbc
  github.com/gravitational/teleport/e/lib/operatortest.TestLoginRuleController.func1()
      /__w/teleport.e/teleport.e/e/lib/operatortest/loginrule_controller_test.go:43 +0x52
  testing.tRunner()
      /opt/go/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/go/src/testing/testing.go:1629 +0x47

Goroutine 95 (running) created at:
  testing.(*T).Run()
      /opt/go/src/testing/testing.go:1629 +0x805
  github.com/gravitational/teleport/e/lib/operatortest.TestLoginRuleController()
      /__w/teleport.e/teleport.e/e/lib/operatortest/loginrule_controller_test.go:41 +0x168
  testing.tRunner()
      /opt/go/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/go/src/testing/testing.go:1629 +0x47

Goroutine 91 (running) created at:
  testing.(*T).Run()
      /opt/go/src/testing/testing.go:1629 +0x805
  github.com/gravitational/teleport/e/lib/operatortest.TestLoginRuleController()
      /__w/teleport.e/teleport.e/e/lib/operatortest/loginrule_controller_test.go:41 +0x168
  testing.tRunner()
      /opt/go/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/go/src/testing/testing.go:1629 +0x47
==================
=== NAME  TestLoginRuleController/update
    testing.go:1446: race detected during execution of test
=== NAME  TestLoginRuleController/deletion_drift
    testing.go:1446: race detected during execution of test
=== NAME  TestLoginRuleController/creation
    testing.go:1446: race detected during execution of test
time="2023-05-22T[17](https://github.com/gravitational/teleport.e/actions/runs/5048568926/jobs/9056926772?pr=1470#step:9:18):[18](https://github.com/gravitational/teleport.e/actions/runs/5048568926/jobs/9056926772?pr=1470#step:9:19):48Z" level=warning msg="Mux serve failed." error="listener is closed\n\taccept tcp 127.0.0.1:40281: use of closed network connection" trace.component=auth
--- FAIL: TestLoginRuleController (0.68s)
    --- FAIL: TestLoginRuleController/update (7.65s)
    --- FAIL: TestLoginRuleController/deletion_drift (7.79s)
    --- FAIL: TestLoginRuleController/creation (8.15s)
FAIL
FAIL	github.com/gravitational/teleport/e/lib/operatortest	9.0[21](https://github.com/gravitational/teleport.e/actions/runs/5048568926/jobs/9056926772?pr=1470#step:9:22)s
FAIL
```

--- a/integrations/operator/controllers/resources/testlib/env.go
+++ b/integrations/operator/controllers/resources/testlib/env.go
@@ -50,16 +50,16 @@ import (
 	"github.com/gravitational/teleport/lib/service/servicecfg"
 )
 
-// Scheme is our own test-specific scheme to avoid using the global
+// scheme is our own test-specific scheme to avoid using the global
 // unprotected scheme.Scheme that triggers the race detector
-var Scheme = apiruntime.NewScheme()
+var scheme = apiruntime.NewScheme()
 
 func init() {
-	utilruntime.Must(v1.AddToScheme(Scheme))
-	utilruntime.Must(resourcesv1.AddToScheme(Scheme))
-	utilruntime.Must(resourcesv2.AddToScheme(Scheme))
-	utilruntime.Must(resourcesv3.AddToScheme(Scheme))
-	utilruntime.Must(resourcesv5.AddToScheme(Scheme))
+	utilruntime.Must(v1.AddToScheme(scheme))
+	utilruntime.Must(resourcesv1.AddToScheme(scheme))
+	utilruntime.Must(resourcesv2.AddToScheme(scheme))
+	utilruntime.Must(resourcesv3.AddToScheme(scheme))
+	utilruntime.Must(resourcesv5.AddToScheme(scheme))
 }
 
 func createNamespaceForTest(t *testing.T, kc kclient.Client) *core.Namespace {
@@ -178,7 +178,7 @@ func (s *TestSetup) StartKubernetesOperator(t *testing.T) {
 	}
 
 	k8sManager, err := ctrl.NewManager(s.K8sRestConfig, ctrl.Options{
-		Scheme:             Scheme,
+		Scheme:             scheme,
 		MetricsBindAddress: "0",
 	})
 	require.NoError(t, err)
@@ -277,7 +277,7 @@ func SetupTestEnv(t *testing.T, opts ...TestOption) *TestSetup {
 	require.NoError(t, err)
 	require.NotNil(t, cfg)
 
-	k8sClient, err := kclient.New(cfg, kclient.Options{Scheme: Scheme})
+	k8sClient, err := kclient.New(cfg, kclient.Options{Scheme: scheme})
 	require.NoError(t, err)
 	require.NotNil(t, k8sClient)
 

--- a/integrations/operator/controllers/resources/testlib/env.go
+++ b/integrations/operator/controllers/resources/testlib/env.go
@@ -28,9 +28,11 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	core "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiruntime "k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/client-go/rest"
-	"k8s.io/kubectl/pkg/scheme"
 	ctrl "sigs.k8s.io/controller-runtime"
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
@@ -47,6 +49,18 @@ import (
 	"github.com/gravitational/teleport/lib/modules"
 	"github.com/gravitational/teleport/lib/service/servicecfg"
 )
+
+// Scheme is our own test-specific scheme to avoid using the global
+// unprotected scheme.Scheme that triggers the race detector
+var Scheme = apiruntime.NewScheme()
+
+func init() {
+	utilruntime.Must(v1.AddToScheme(Scheme))
+	utilruntime.Must(resourcesv1.AddToScheme(Scheme))
+	utilruntime.Must(resourcesv2.AddToScheme(Scheme))
+	utilruntime.Must(resourcesv3.AddToScheme(Scheme))
+	utilruntime.Must(resourcesv5.AddToScheme(Scheme))
+}
 
 func createNamespaceForTest(t *testing.T, kc kclient.Client) *core.Namespace {
 	ns := &core.Namespace{
@@ -164,7 +178,7 @@ func (s *TestSetup) StartKubernetesOperator(t *testing.T) {
 	}
 
 	k8sManager, err := ctrl.NewManager(s.K8sRestConfig, ctrl.Options{
-		Scheme:             scheme.Scheme,
+		Scheme:             Scheme,
 		MetricsBindAddress: "0",
 	})
 	require.NoError(t, err)
@@ -263,19 +277,7 @@ func SetupTestEnv(t *testing.T, opts ...TestOption) *TestSetup {
 	require.NoError(t, err)
 	require.NotNil(t, cfg)
 
-	err = resourcesv1.AddToScheme(scheme.Scheme)
-	require.NoError(t, err)
-
-	err = resourcesv5.AddToScheme(scheme.Scheme)
-	require.NoError(t, err)
-
-	err = resourcesv2.AddToScheme(scheme.Scheme)
-	require.NoError(t, err)
-
-	err = resourcesv3.AddToScheme(scheme.Scheme)
-	require.NoError(t, err)
-
-	k8sClient, err := kclient.New(cfg, kclient.Options{Scheme: scheme.Scheme})
+	k8sClient, err := kclient.New(cfg, kclient.Options{Scheme: Scheme})
 	require.NoError(t, err)
 	require.NotNil(t, k8sClient)
 

